test/core/client_channel/lb_policy/pick_first_test.cc

```diff
@@ -177,6 +269,61 @@ TEST_F(PickFirstTest, GoesIdleWhenConnectionFailsThenCanReconnect) {
   }
 }
 
+TEST_F(PickFirstTest, WithShuffle) {
+  testing::ScopedExperimentalEnvVar env_var(
+      "GRPC_EXPERIMENTAL_PICKFIRST_LB_CONFIG");
+  // 6 addresses have 6! = 720 permutations or 0.1% chance that the shuffle
+  // returns initial sequence
+  constexpr std::array<absl::string_view, 6> kAddresses = {
+      "ipv4:127.0.0.1:443", "ipv4:127.0.0.1:444", "ipv4:127.0.0.1:445",
+      "ipv4:127.0.0.1:446", "ipv4:127.0.0.1:447", "ipv4:127.0.0.1:448"};
+  absl::Status status = ApplyUpdate(
+      BuildUpdate(kAddresses, MakePickFirstConfig(true)), lb_policy_.get());
+  EXPECT_TRUE(status.ok()) << status;
+  std::vector<absl::string_view> prev_attempt_connect_order;
+  // There is 0.1% chance this check fails by design. Not an assert to prevent
+  // flake
+  if (absl::MakeConstSpan(prev_attempt_connect_order) ==
+      absl::MakeConstSpan(kAddresses)) {
+    gpr_log(GPR_INFO, "Address order did not change");
+  }
+  constexpr size_t kMaxAttempts = 5;
+  bool shuffled = false;
+  for (size_t attempt = 0; attempt < kMaxAttempts && !shuffled; ++attempt) {
```

Comment:
Instead of checking `!shuffled` here, please add a `break` on line 297 below.

Reply:
Done